health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java

```diff
@@ -56,8 +57,10 @@
    * Other metadata fields on the HealthDto will also be populated.
    */
   public HealthDto doPrimaryHealthCheck() {
-    // TODO - allow some dependencies to be marked as 'secondary', and skip those here
-    return checkHealth(healthDependencies.values());
+    List<HealthDependency> primaryHealthDependencies = healthDependencies.values().stream()
+        .filter(HealthDependency::getPrimary)
+        .collect(Collectors.toList());
```

Comment:
Is there a way to avoid converting this back into a list? The `checkHealth` method really just needs to be able to iterate over the dependencies and doesn't actually need it to be a list.

It's likely a very small list right now, so it probably doesn't matter for performance. If it's easy to do, then we should change this, but I'm not sure the best way while using streams.

cc: @nathanschile

Reply:
I'm not sure if there's a way we can do it and preserve the streaming. The basic tenants of streaming in java is that we filter a set of values by a condition and the collect them at the end. 

We might be able to do it by turning it into an iterator and then turning that into an iterable but that seems like a bad solution. 

`healthDependencies.values().stream()
        .filter(HealthDependency::getPrimary).iterator();`

Then,

`Iterable<HealthDependency> iterable = () -> primaryHealthDependencies;`

http://www.lambdafaq.org/how-can-i-turn-an-iterator-into-an-iterable/

Reply:
I saw that post as well. That doesn't read as cleanly as what we have now.

I think the best approach would be to refactor `checkHealth` completely so that we could use a `map` function on the stream we already have here to dependency DTOs. However, I don't think it's worth the effort on this PR.